### PR TITLE
Review comments implemented

### PR DESCRIPTION
**Pull request by Vipul6**

### CODE REVIEW

File: lib/components/molecules/StickyHeader/StickyHeader.js [hunk 1]
```diff
@@ -0,0 +1,39 @@
...
+import styled from 'styled-components';
+import { Props } from './types/index';
+
+import styles from './StickyHeader.style';
+
+const StickyHeader = ({ children, className }: Props): Node => {
+  const [isSticky, setSticky] = useState(false);
+  const wrapperRef = useRef(null);
+
+  const debounce = (func, delay) => {
```
Comment (vinodloha) [inline lib/components/molecules/StickyHeader/StickyHeader.js]: Why not create debounce as custom hook?

File: lib/components/molecules/StickyHeader/StickyHeader.style.js [hunk 2]
```diff
@@ -0,0 +1,10 @@
+import { css } from 'styled-components';
+
+const styles = css`
+  header {
```
Comment (vinodloha) [inline lib/components/molecules/StickyHeader/StickyHeader.style.js:4]: Have you tested this in all browsers that 
- Sticky Behaviour is consistent. 
- Page not causing a flicker on content below header component when it becomes sticky.

File: lib/components/molecules/StickyHeader/tests/StickyHeader.test.js [hunk 3]
```diff
@@ -0,0 +1,25 @@
+import React from 'react';
+import { shallow } from 'enzyme';
+import { StickyHeaderVanilla } from '../StickyHeader';
+
+describe('Test render <StickyHeader /> Component', () => {
```
Comment (vinodloha) [inline lib/components/molecules/StickyHeader/tests/StickyHeader.test.js:5]: These tests are not enough, they are not even testing the core functionality

File: lib/components/molecules/StickyHeader/types/index.js [hunk 4]
```diff
@@ -0,0 +1,7 @@
+// @flow
+import type, { Node } from 'react';
+
+export type Props = {
```
Comment (vinodloha) [inline lib/components/molecules/StickyHeader/types/index.js:5]: Your component works with a state, isSticky we should define that in types too

File: lib/components/molecules/StickyHeader/StickyHeader.js [hunk 5]
```diff
@@ -0,0 +1,39 @@
...
+    };
+  };
+
+  const handleScroll = () => {
+    if (wrapperRef.current != null) {
+      setSticky(window.scrollY >= wrapperRef.current.offsetTop);
+    }
+  };
+
+  useEffect(() => {
```
Comment (vinodloha) [inline lib/components/molecules/StickyHeader/StickyHeader.js:13]: Scroll event handling could also be a custom Hook

File: lib/components/molecules/StickyHeader/StickyHeader.js [hunk 6]
```diff
@@ -0,0 +1,39 @@
...
+    window.addEventListener('scroll', handleScroll);
+    return () => {
+      window.removeEventListener('scroll', () => handleScroll);
+    };
+  });
+
+  return (
+    <div className={`${isSticky ? className : ''}`} ref={wrapperRef}>
```
Comment (vinodloha) [inline lib/components/molecules/StickyHeader/StickyHeader.js]: How would a disabled user will know about sticky behaviour?

File: lib/components/molecules/StickyHeader/types/index.js (hunk 4 above)
Comment (Vipul6) [reply lib/components/molecules/StickyHeader/types/index.js:5]: Fixed.

File: lib/components/molecules/StickyHeader/StickyHeader.js (hunk 5 above)
Comment (Vipul6) [reply lib/components/molecules/StickyHeader/StickyHeader.js:13]: Fixed.

File: lib/components/molecules/StickyHeader/StickyHeader.js (hunk 6 above)
Comment (Vipul6) [reply lib/components/molecules/StickyHeader/StickyHeader.js]: I have given it role"Sticky Header'.

File: lib/components/molecules/StickyHeader/tests/StickyHeader.test.js (hunk 3 above)
Comment (Vipul6) [reply lib/components/molecules/StickyHeader/tests/StickyHeader.test.js:5]: Facing issue in hook testing. Will discuss it.

**Pull request closed without merging**